partner_contact_age_range/models/res_partner.py

```diff
@@ -0,0 +1,34 @@
+# Copyright (C) 2019-Today: Druidoo (<https://www.druidoo.io>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    age_range_id = fields.Many2one(
+        "res.partner.age.range",
+        "Age Range",
+        compute="_compute_age_range_id",
+        store=True,
+    )
+
+    @api.depends("age")
+    def _compute_age_range_id(self):
+        age_ranges = self.env["res.partner.age.range"].search([])
+        for record in self:
```

Comment:
Put in a continue if there is no age (because there might not be a birthdate).

Reply:
@NL66278 done

Reply:
Actually, if there's no age we need to set the age_range_id to False, otherwise Odoo will not update the previous value

ie: let's say you remove the birthday of a partner, you expect the age_range to be removed as well

Reply:
@ajayjaviya @ivantodorovich is right. So in sum this method should be IMHO like this:
```
    @api.depends("age")
    def _compute_age_range_id(self):
        age_ranges = self.env["res.partner.age.range"].search([])
        for record in self:
            age_range = record.age_range_id
            if not record.age:
                age_range = False
            else:
                age_range = age_ranges.filtered(
                    lambda age_range: age_range.age_from <= record.age <= age_range.age_to
                )
            if record.age_range_id != age_range:
                record.age_range_id = age_range.id
```

Reply:
btw @NL66278 , are you completely sure that we need to check if the value has changed before assignation?
I mean, isn't the ORM smart enough to avoid writing on the record if the field hasn't actually changed?

I'm asking because I honestly don't know :sweat_smile:

Reply:
@ivantodorovich You can check it in a shell session (I use bin/python_odoo from buildout):
```
>>> partner = partner_model.browse(8)
>>> partner.write_date
'2020-02-04 09:04:37'
>>> partner.customer
True
>>> partner.customer = True
>>> partner.write_date
'2020-05-07 11:58:27'
```

Reply:
@NL66278  and @ivantodorovich  done